nncf/quantization/algorithms/accuracy_control/algorithm.py

```diff
@@ -0,0 +1,326 @@
+                               fast_bias_correction, model_type, ignored_scope)
+
+    # Backends
+    backend = get_backend(model)
+    algo_backend = get_algo_backend(backend)
+
+    initial_metric = validation_fn(algo_backend.prepare_for_inference(model),
+                                   validation_dataset.get_data())
+    nncf_logger.info(f'Metric of initial model:   {initial_metric}')
+
+    quantized_metric = validation_fn(algo_backend.prepare_for_inference(quantized_model),
+                                     validation_dataset.get_data())
+    nncf_logger.info(f'Metric of quantized model: {quantized_metric}')
+
+    return restore_accuracy(model, initial_metric,
+                            quantized_model, quantized_metric,
+                            validation_dataset, validation_fn, max_drop)
+
+
+def _create_message(nodes: Iterable[NNCFNode]) -> str:
+    names = [f'\t{x.node_name}' for x in nodes]
+    return '\n'.join(names)
+
+# pylint: disable=R0912
+# pylint: disable=R0915
+def restore_accuracy(initial_model: TModel,
+                     initial_metric: float,
+                     quantized_model: TModel,
+                     quantized_metric: float,
+                     validation_dataset: Dataset,
+                     validation_fn: Callable[[Any, Iterable[Any]], float],
+                     max_drop: float = 0.01) -> TModel:
+    """
+    Restores the accuracy of the quantized model by removing the groups of quantizers
+    that contribute the most to the drop in accuracy.
+
+    :param initial_model: Initial model (not quantized).
+    :param initial_metric: Metric value for initial model.
+    :param quantized_model: Quantized model.
+    :param quantized_metric: Metric value for quantized model.
+    :param validation_dataset: A dataset for the validation process.
+    :param validation_fn: A validation function to validate the model. It should take
+        two argumets:
+        - `model`: model to be validate.
+        - `validation_dataset`: dataset that provides data items to
+              validate the provided model.
+        The function should return the value of the metric with the following meaning:
+        A higher value corresponds to better performance of the model.
+    :param max_drop: The maximum absolute accuracy drop that should be achieved.
+    :return: The quantized model whose metric `final_metric` is satisfied the following condition
+
+        initial_metric - final_metric <= max_drop.
+    """
+    accuracy_drop = initial_metric - quantized_metric
+    nncf_logger.info(f'Accuracy drop:             {accuracy_drop}')
+
+    if accuracy_drop <= max_drop:
+        return quantized_model
+
+    nncf_logger.info('Changing the scope of quantizer nodes.')
+
+    # Hyperparameters of algorithm
+    RANKING_SUBSET_SIZE = 300
+    MAX_NUM_ITERATIONS = sys.maxsize
+    USE_PREVIOUS_IF_DROP_INCREASE = True
```

Comment:
> Why these parameters are not a part of the algorithm options?

Good question. We need to decide which options should be provided to user.

> Why is the RANKING_SUBSET_SIZE not connected with subset_size at least?

These are two different parameters. There are no dependencies between them.
- `subset_size` (calibration subset size) - number of data items that will be taken from the `calibration_dataset` to calibrate the quantized model.
- `ranking_subset_size` - number of data items that will be taken from the `validation_dataset` to rank groups of quantizers.

Reply:
There still is no condition that regulates `RANKING_SUBSET_SIZE` depending on the `validation_dataset` size. And it is unclear what will happen in case if the length of the dataset will be less than length of ranking subset size.

Reply:
Could you please look at the `get_ranking_subset_indices()` method implementation?

Reply:
Great, now I can see that everything will be correct.
But the problem is that it's not clear from the first view, that somewhere in the `rank_quantizers` that accepts `RANKING_SUBSET_SIZE` as one of the input variables, the correct number of the ranking subset will be calculated.

Reply:
The docstring for the `rank_quantizers()` method was added. It should be clear now.